core-war/src/main/java/org/silverpeas/web/directory/control/DirectorySessionController.java

```diff
@@ -452,21 +454,36 @@ public DirectoryItemList getLastListOfUsersCalled() {
   }
 
   /**
-   * return All users of Space who has Id="spaceId"
-   * @param spaceId:the ID of Space
-   *
+   * Gets all users of space according to given parameters.
+   * @param spaceId the unique identifier of a space.
+   * @param componentInstanceRolesLookup true to also lookup roles of component instances of the
+   * space and its sub-spaces, false to take into account only roles of the specified space.
    */
-  public DirectoryItemList getAllUsersBySpace(String spaceId) {
+  public DirectoryItemList getAllUsersBySpace(String spaceId,
+      final boolean componentInstanceRolesLookup) {
```

Comment:
In fact here the parameter  `componentInstanceRolesLoojkup` has a technical meaning here. It should be named either as `onlyInSpace` (the inverse of the meaning of `componentInstanceRolesLookup`) or  like `inWholeSpaceTree` or `notOnlyInSpace` (in the meaning of the parameter). In my advice, such name is more understandable: get all users only in the space or get all users in the whole space tree. This can be also split into two dedicated public methods that delegates the call to a private method with the boolean named as you want (it is a technical meaning and hence it is kept private)

Reply:
Yes. I will use your `inWholeSpaceTree` suggestion.